### DISCUSSION TITLE
EVTV ESP32 CANDue troubleshooting?

### DISCUSSION THREAD

**Discussion by cgalpin** [collin80/SavvyCAN#670] [General]
Collin, and anyone else who might be able to help, I am trying to figure out whats wrong with my ESP32 CANDue board. I have had it for several years and it used to work, but have not used it in a while.  I went to go use it last week and have not been able to get it to work, even on a test setup i know it used to work on - a leaf battery where I can get the data from it via leafspy plugged into a OBD2 port on the same bus so i know there is traffic.

So it connects to the board fine and I can see console serial communication just fine, but no matter what bus or speed i try (and even with and without additional 120ohm termination)  i can't get any CAN messages on the main window. I ran the ESP32RET firmware updater just in case, but no better.

How can I troubleshoot this? I suspect hardware, but would prefer to verify before just buying another.

tia,
charles

**Comment by collin80** [comment]
Well, probably the best advice is to scope the output of the transceiver with an oscilloscope but that presupposes that you have one. My best guess would be a bad transceiver. Those are cheaper than a whole new board but would require you could desolder an 8 pin module then solder down the new one.

The easiest way to test a dual CAN board like that is to run short wires H to H and L to L between the two CAN buses. Then you can send traffic between them and test that they both work. This requires no extra hardware other than the two short wires. Just "CANSEND0=0x123,4,1,10,20,30" or something like that. If the serial console shows that CAN1 got a frame then they both work. If it doesn't then you're back to the above where you've got to see if one or both transceivers have failed.

One last idea to check: it should have defaulted to the right board but make sure it doesn't think you have an A0 or A5 board. I think systype should be 1 for the EVTV ESP32 board. But, it doesn't hurt to check that.

**Comment by simotronic** [comment]
Hi,
You can also use a digital DVM. When the bus is running you'll get some AC
voltage, depending on the data rate. When it's idle, it should be 2.5 VDC.
When the bus is asserted, the value is 0 VDC.
Also make sure the H & L lines are the right way round and the ground is
connected. Ideally a scope is needed.

**Comment by cgalpin** [comment]
@collin, I don't have an oscilloscope so I tried wiring the two busses. Do I send the "CANSEND0=0x123,4,1,10,20,30" using the "Send:" field and clicking "Send Text" button under the console? I did not get anything in the console other than the serial comms i see all the time.

How do I check the systype?

@simtronic, with the short wires connecting the two ports I have 0V between H/L lines. I do not have a ground in this setup - not sure what to ground against in this test (i did connect ground to one of the ground pins but no change). 

thanks,
charles

**Comment by collin80** [comment]
Yes, sending the command CANSEND0=0x123,4,1,10,20,30 should cause it to send a frame out of CAN0 and you should then get a message that CAN1 received a message. You should get confirmation in both directions so if you didn't get any response then try again. If that doesn't work then make sure you have some sort of line ending set (CR, LF, or CRLF).

You can check the systype with the ? command. Once again, you need it to send a line ending but otherwise just ?

Your board should have a ground pin near the CAN wires. Use that even if nothing is attached there. It's still a ground. Measure from ground to the H and L lines. They should be floating around 2 to 2.5v. If there is traffic on them you will measure voltage between them too. If there is no traffic then the voltage between H and L is zero.

**Comment by cgalpin** [comment]
Oh, i was trying in SavvyCAN itself. I switched to a straight serial connection and was able to send the commands. Systype is 1.

When I send CANSEND0=0x123,4,1,10,20,30 all i see is

Sending frame with id: 0x123 len: 4
!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!!!!!

I get ~2.8v from ground to each can line, but see no voltage delta across H/L when running the command (but it's only one packet).

I have ordered another board, but would still like to try figure out what's wrong with this one, and fix it if I can.

thanks

p.s. I sent the command a few times. Here are the entire contents of the serial console after connecting, sending ?, and then the CANSEND0 commands

Build number: 618
System Menu:

Enable line endings of some sort (LF, CR, CRLF)

Short Commands:
h = help (displays this message)
R = reset to factory defaults
s = Start logging to file
S = Stop logging to file

Config Commands (enter command=newvalue). Current values shown in parenthesis:

SYSTYPE=1 - Set board type (0=Macchina A0, 1=EVTV ESP32 Board 2=Macchina A5)
LOGLEVEL=1 - set log level (0=debug, 1=info, 2=warn, 3=error, 4=off)

CANEN0=1 - Enable/Disable CAN0 (0 = Disable, 1 = Enable)
CANSPEED0=500000 - Set speed of CAN0 in baud (125000, 250000, etc)
CANLISTENONLY0=0 - Enable/Disable Listen Only Mode (0 = Dis, 1 = En)

CANSEND0=ID,LEN,<BYTES SEPARATED BY COMMAS> - Ex: CAN0SEND=0x200,4,1,2,3,4

CANEN1=1 - Enable/Disable CAN1 (0 = Disable, 1 = Enable)
CANSPEED1=500000 - Set speed of CAN1 in baud (125000, 250000, etc)
CANFDRATE1=5000000 - FD Data Speed of CAN1 (500000, 2000000, etc)
CANFDMODE1=0 - Allow FD traffic on CAN1 (0 = Disable, 1 = Enable)
CANLISTENONLY1=0 - Enable/Disable Listen Only Mode (0 = Dis, 1 = En)

CANSEND1=ID,LEN,<BYTES SEPARATED BY COMMAS> - Ex: CAN0SEND=0x200,4,1,2,3,4

BINSERIAL=0 - Enable/Disable Binary Sending of CANBus Frames to Serial (0=Dis, 1=En)

BTMODE=0 - Set mode for Bluetooth (0 = Off, 1 = On)
BTNAME=ELM327-ESP32RET - Set advertised Bluetooth name

LAWICEL=1 - Set whether to accept LAWICEL commands (0 = Off, 1 = On)

WIFIMODE=2 - Set mode for WiFi (0 = Wifi Off, 1 = Connect to AP, 2 = Create AP
SSID=ESP32RETSSID - Set SSID to either connect to or create
WPA2KEY=aBigSecret - Either passphrase or actual key
Sending frame with id: 0x123 len: 4
!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!**!!!!!!Sending frame with id: 0x123 len: 4
Sending frame with id: 0x123 len: 4
Sending frame with id: 0x123 len: 4

**Comment by cgalpin** [comment]
The board is definitely bad - with a new one, i can read the CAN messages from my leaf battery no problem. Are the transceivers you mention labelled U2 and U5?

I do have a related question though - i'm trying to get CAN messages from a different battery (hyundai ioniq 5) and am not having any luck. One thing I noticed is the serial comms show in the SavvyCAN serial console changes when I apply 12v to the ignition wire of the BCM. Can you explain what this is and why the serial messages would change, but no CAN messages are shown?

tia,
charles

before IGN on:

Write to serial -> f1 9 
Write to serial -> f1 9 
Got data from serial. Len = 4
Got data from serial. Len = 4
f1 09 de ad 
f1 09 de ad 
Write to serial -> f1 9 
Write to serial -> f1 9 
Got data from serial. Len = 4
Got data from serial. Len = 4
f1 09 de ad 
f1 09 de ad 
Write to serial -> f1 9 
Write to serial -> f1 9 
Got data from serial. Len = 4
Got data from serial. Len = 4
f1 09 de ad 
f1 09 de ad 

After IGN on:

Got data from serial. Len = 56
Got data from serial. Len = 56
21 21 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 
21 21 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 
Got data from serial. Len = 64
Got data from serial. Len = 64
2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 
2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 
Got data from serial. Len = 56
Got data from serial. Len = 56
2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 21 21 21 21 21 21 21 21 
2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 21 21 21 21 21 21 21 21 
Got data from serial. Len = 32
Got data from serial. Len = 32
21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 
21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 21 
Got data from serial. Len = 50
Got data from serial. Len = 50
21 21 2a 2a 21 21 21 21 21 21 21 21 21 21 21 21 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 
21 21 2a 2a 21 21 21 21 21 21 21 21 21 21 21 21 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 
Got data from serial. Len = 64
Got data from serial. Len = 64
2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 
2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 
Got data from serial. Len = 64
Got data from serial. Len = 64
2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 
2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 
Got data from serial. Len = 64
Got data from serial. Len = 64
2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 
2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 
Got data from serial. Len = 64
Got data from serial. Len = 64
2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 
2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 
Write to serial -> f1 9 
Write to serial -> f1 9 
Got data from serial. Len = 68
Got data from serial. Len = 68
2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 f1 09 de ad 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 
2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 f1 09 de ad 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 
Got data from serial. Len = 64
Got data from serial. Len = 64
2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 
2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21 2a 2a 21 21

**Comment by collin80** [comment]
Yes U2 and U5 are the transceivers.

0x21 and 0x2A are ! and * and are sent when CAN module errors are detected. Either the CAN speed is wrong, it's using CAN-FD traffic, or otherwise something is wrong at the bus level.

**Reply by cgalpin** [reply]
Ok, so it looks like this uses CAN-FD. The status of CAN-FD support is not clear to me. Should I be able to read CAN-FD messages or not?

**Reply by collin80** [reply]
Well, yes, it should work to read CAN-FD frames. But, support is not super well tested at this point. I've had nothing really that does CAN-FD. Well, at least I didn't used to. I have a 2022 Bolt EUV and it might use CAN-FD but I don't think so. I think anywhere they would have needed the extra capacity they went with FlexRAY instead. So, in theory CAN-FD should work but I haven't been able to test it well. One thing that is definitely true is that you cannot set the FD speed in SavvyCAN. You have to do that on the serial console directly in the RET device itself. And, even then, I've found that certain combinations seem to work oddly. I think I couldn't get CAN-FD working properly when using 500k for the nominal speed. I think it worked at 1M nominal and then 2M, 3M, 4M for the FD data bytes speed. I never figured out why that is. It seems to be some oddity with the way the CAN library is doing timing. But, once again, I had nothing to test against.

**Reply by cgalpin** [reply]
Ok, I believe I have a test case - a Hyundai ioniq 5 battery. If you are interested in giving it a try I could send you the battery ecu (as long as you send it back when done :) ).

But trying this myself I can't even enable FD on the bus connection. Maybe i just don't know what I am doing. I have GVRET connection over the serial port to the CANDue but the FD option is not visible on either bus.

Another person on openinvertor.org points out that it's likely this line of code, which requires a SERIALBUS connection type (and gvet is type GVRET_SERIAL

https://github.com/collin80/SavvyCAN/blob/1999da88c5d6b32bd432b737ba1b60d1a15ff47b/connections/connectionwindow.cpp#L401C19-L401C19

That discussion is here: https://openinverter.org/forum/viewtopic.php?p=63711&sid=243189d09041f7ceb7e404a287d714a8#p63711

thoughts?

tia,
charles

**Reply by danjhunter** [reply]
I am working on an application as well with CANFD at 2MBit/sec. I see a stream of messages via cool term but they also contain a significant about of ! & *. I can see CANFD frames and CAN2.0 frames. @collin80 any idea how to fix this? Thanks

**Comment by cgalpin** [comment]
Thanks. I was using the CAN-FD bus because I read somewhere they used can-fd so that’s a possibility. Can savvycan not handle those?Since is did see some messages from a obd2 device connected, I am assuming I had the speed right. -- charlesOn Oct 2, 2023, at 7:26 PM, Collin Kidder ***@***.***> wrote:﻿
Yes U2 and U5 are the transceivers.
0x21 and 0x2A are ! and * and are sent when CAN module errors are detected. Either the CAN speed is wrong, it's using CAN-FD traffic, or otherwise something is wrong at the bus level.

**Comment by mitchdetailed** [comment]
chevy went all in with FD on globalB, haven't seen a single Gm application
utilizing flexray..

**Reply by collin80** [reply]
Well, now you've heard of one. ;) The Bolt EUV really uses CAN and FlexRay. The object detection system uses FlexRay. I have the schematics for the car. I was quite surprised at this too. But, it could be because it's a bastardized car. Normally only GlobalB cars have super cruise. But, SC is possible on the EUV, apparently by using FlexRay. It's... odd... They then decided to build the car for only two model years and discontinue it. Now everything is GlobalB, probably including the Bolt when it comes back in 2025.

**Comment by cgalpin** [comment]
@collin80 would you mind providing a link to the right component to buy for the U2 and U5 transceivers for a esp32 can_due? I have been meaning to buy these but don't want to buy the wrong thing. If both can be the MCP2517FD, I'm happy to have both be CAN-FD capable once replaced.

tia,
charles

**Comment by weisheimer** [comment]
Hi Charles,

My EVTV V2 ESP32 CANDue has the original TI SN65HVD234DR CAN transceiver in U2
It is available (for now, they are obsoleting this chip so quantities will be limited) at:
https://www.digikey.com/en/products/detail/texas-instruments/SN65HVD234DR/1574511
For U5, there is the MicroChip FD chip, MCP2562FD transceiver which is available at:
https://www.digikey.com/en/products/detail/microchip-technology/MCP2562FD-E-SN/4842808
I do not believe you can make both FD by replacing U2 with the MCP2562FD transceiver as it currently stands, but check to see if this is the version that you have and if these are the chips on your board.
You may not need this tip, but what I do is to cut the leads at the body with a sharp exacto knife and then use a small soldering iron with solder wick to remove the old pins, then solder in the new parts.
Collin can elaborate further with better info, but I wanted to post what I found on my boards purchased in 2018.
Good luck and let us know how it goes!
Mark Weisheimer

**Comment by cgalpin** [comment]
Hi Mark!

This is very helpful, thanks. I do have a V2 board (i should have mention that when asking, sorry). My eyesight is terrible, but if you zoom into the attached photo, you can see the U5 matches, but not the U2. I *think* my U2 says VP23498M AF5Z. Maybe a TI chip, so the same? Let me know what you think please.

Thanks for the tip, i am sure it will come in handy getting the originals off.

thanks again,
charles
![esp32 candue transceivers](https://github.com/collin80/SavvyCAN/assets/2838085/6b639cc5-190d-48a4-b480-6229fb0bf545)

**Reply by weisheimer** [reply]
Hi Charles,

U2 on the ESP32 CANDue here is marked the same, VP234 with slightly different date code of 834M here.
That is the marking for the TI SN65HVD234DR as shown in the Datasheet:
https://www.ti.com/lit/ds/symlink/sn65hvd234.pdf?ts=1702161128275&ref_url=https%253A%252F%252Fwww.ti.com%252Fproduct%252Fde-de%252FSN65HVD234
See page 5 for the device marking info.
Best
Mark

**Comment by cgalpin** [comment]
That's awesome thanks!
